File: Prompting-and-Iterative-Refinement-Assignment-Bolierplate-Code/python/benchmark.py
# benchmark.py
import csv
import time
from log_analyzer import get_top_slowest_endpoints

def load_logs(filename="../api_logs.csv"):
    logs = []
    with open(filename, mode='r', encoding='utf-8') as f:
        reader = csv.DictReader(f)
        for row in reader:
            logs.append({
                'url': row['url'],
                'status': int(row['status']),
                'latency_ms': int(row['latency_ms'])
            })
    return logs

if __name__ == "__main__":
    print("Loading logs from CSV...")
    mock_data = load_logs()
    print(f"Successfully loaded {len(mock_data)} logs.")
    
    print("Running baseline analyzer... (This might take 10-20 seconds!)")
    start_time = time.time()
    
    results = get_top_slowest_endpoints(mock_data)
    
    end_time = time.time()
    
    print("\n--- TOP 5 SLOWEST ENDPOINTS ---")
    for res in results:
        print(f"{res['endpoint']}: {res['avg_latency']:.2f} ms")
        
    print(f"\nExecution Time: {end_time - start_time:.4f} seconds")

File: Prompting-and-Iterative-Refinement-Assignment-Bolierplate-Code/java/Benchmark.java
// Benchmark.java
import java.io.*;
import java.util.*;

public class Benchmark {

    public static void main(String[] args) {
        System.out.println("Loading logs from CSV...");
        List<Map<String, Object>> mockData = new ArrayList<>();
        
        try (BufferedReader br = new BufferedReader(new FileReader("../api_logs.csv"))) {
            String line = br.readLine(); // Skip header
            while ((line = br.readLine()) != null) {
                String[] parts = line.split(",");
                if (parts.length == 3) {
                    Map<String, Object> log = new HashMap<>();
                    log.put("url", parts[0]);
                    log.put("status", Integer.parseInt(parts[1]));
                    log.put("latency_ms", Integer.parseInt(parts[2]));
                    mockData.add(log);
                }
            }
        } catch (IOException e) {
            System.err.println("Error reading CSV file: " + e.getMessage());
            return;
        }

        System.out.println("Successfully loaded " + mockData.size() + " logs.");
        System.out.println("Running baseline analyzer... (This might take 10-20 seconds!)");

        long startTime = System.currentTimeMillis();
        List<Map<String, Object>> results = LogAnalyzer.getTopSlowestEndpoints(mockData);
        long endTime = System.currentTimeMillis();

        System.out.println("\n--- TOP 5 SLOWEST ENDPOINTS ---");
        for (Map<String, Object> res : results) {
            System.out.printf("%s: %.2f ms\n", res.get("endpoint"), res.get("avg_latency"));
        }
        
        System.out.println("\nExecution Time: " + (endTime - startTime) / 1000.0 + " seconds");
    }
}
